ls1intum/Artemis
# Do not allow tutors to override their own assessments if there is a complaint

### Checklist
- [x] Server: I implemented the changes with good performance and prevented too many database calls

### Motivation and Context
Fixes #1922
**Edit:** We want to allow the tutor to still see the complaint. However, he/she should not obtain the lock or be able to override the result.

### Description
- [x] Remove transactional & refactor getAllModelingSubmissionsByTutorForExercise
- [x] Remove transactional & refactor getAllTextSubmissionsByTutorForExercise
- [x] Remove transactional & refactor getAllFileUploadSubmissionsByTutorForExercise
- [x] Delete obsolete DB call from ResultRepository and move to SubmissionRepository
- [x] Add a text which lets the tutor know why he cannot override the assessment.
- [x] Make all input elements readOnly, if the tutors assessment has a complaint

### Steps for Testing
1. Create an exam
2. participate and submit it as a student
3. Assess an exercise as a tutor
4. Make a complaint about the assessment as a student
5. As the tutor navigate back to tutor dashboard: You should still see the submission on under your assessed submissions. But you should not be able to obtain the lock and override the assessment. Furthermore, all input fields should be read-only. **Caution: Instructors are always allowed to override**
6. Log in as a different tutor, you should see the submission in your complaints table of the tutor exercise dashboard. You should be able to handle the complaint. The submissions stays in the complaints table also after the complaint has been handled.

### Screenshots
![image](https://user-images.githubusercontent.com/15110960/88105577-6f43a280-cba4-11ea-92af-fc65c2c358e5.png)
